jerry-core/ecma/builtin-objects/ecma-builtin-array.c

```diff
@@ -499,7 +500,24 @@ ecma_builtin_array_dispatch_construct (const ecma_value_t *arguments_list_p, /**
 {
   JERRY_ASSERT (arguments_list_len == 0 || arguments_list_p != NULL);
 
+#if !ENABLED (JERRY_ES2015)
   return ecma_op_create_array_object (arguments_list_p, arguments_list_len, true);
+#else /* ENABLED (JERRY_ES2015) */
+  ecma_value_t result =  ecma_op_create_array_object (arguments_list_p, arguments_list_len, true);
+
+  ecma_object_t *proto_p = ecma_op_get_prototype_from_constructor (JERRY_CONTEXT (current_new_target),
+                                                                    ECMA_BUILTIN_ID_ARRAY_PROTOTYPE);
+  if (proto_p == NULL)
```

Comment:
This check should be performed after `ecma_op_get_prototype_from_constructor` and before `ecma_op_create_array_object `;